### PR TITLE
Added feature for payment gateways banner

### LINKED ISSUE #32414: A/B Test for Banner in Payment Settings
Description: Introduce an A/B test containing a banner for WooCommerce Payments above the settings table.

We should treat it as a nice to have for WC 6.5.

Acceptance criteria:

- [ ] Test a variant where we include a banner about WCPayments above the table.
- [ ] The banner should only appear if the WooCommerce Payment plugin is installed but not set up, and in eligible country.
- [ ] Get Started” should link to the same page as the “Get Started” button in the Payment Methods list.
- [ ] The banner is controlled from an ExPlat experiment named `woocommerce_payments_setting_banner_${YEAR}_${MONTH}`.

Figma: e428PXC0QyEwYqtX48lLq0-fi-3%3A47

[illustration(png)](https://user-images.githubusercontent.com/3747241/160840470-33123389-b6e8-453e-81dd-8a0c51e37733.png)

[illustration(SVG)](https://user-images.githubusercontent.com/3747241/160840510-8c31f63b-efba-468f-83e7-9c3f731cfd0f.svg)

### PR DESCRIPTION
**Pull request by rjchow**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR makes a set of changes in order to implement the feature for a payment gateways banner.

There were two possible approaches to implement this - fully in PHP and the other to embed a React tree on a new element that is introduced on the PHP side.

The decision was made to embed a React tree for ease of future maintenance, and also to make the new mount point an experimental Slotfill so that we can add future items to it.

Explat post: pbxNRc-1yD-p2

Closes #32414 .

###### Control:

image

###### Treatment: 

image

### How to test the changes in this Pull Request:

There are 5 scenarios to test:
- WCPay eligible and WCPay installed and setup completed and assigned to 'treatment' -> Should not see the banner
- WCPay eligible and WCPay not installed -> Should not see the banner
- WCPay eligible and WCPay installed but setup incomplete and assigned to 'treatment' -> Should see the banner
- WCPay eligible and WCPay installed but setup incomplete and assigned to 'control' -> Should not see the banner
- WCPay ineligible and/or WCPay not installed -> Should not see the banner

1. Set up a fresh install of Wordpress with WooCommerce, and depending on the scenario you are testing, install WooCommerce Payments plugin (either as part of onboarding or otherwise)
2. Install and activate [wc-test-helper](https://github.com/woocommerce/woocommerce-admin-test-helper/releases/tag/v0.7.2) plugin
3. Navigate to Tools -> WCA Test Helper
4. Click **Experiments**
5. Toggle the experiment assignment for 'woocommerce_payments_settings_banner_2022_04' to control or treatment depending on the scenario
6. Visit WooCommerce > Settings, Payments tab, and check if the banner is visible or not 

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: I think this is meant to be a dynamic experiment? If so, we should use dynamic month & year

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (chihsuan) [inline plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: I think we could use `getAdminLink` here:

https://github.com/woocommerce/woocommerce/blob/feba6a8dcd55d4f5c7edc05478369c76df082293/plugins/woocommerce-admin/client/tasks/fills/products/products.js#L64-L66

File: plugins/woocommerce-admin/client/payments-welcome/index.tsx
Comment (chihsuan) [inline plugins/woocommerce-admin/client/payments-welcome/index.tsx:33]: Nice work for refactoring this!

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (chihsuan) [inline plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: Looks like `payment-recommendations.tsx` also has the same code. How about creating a util function for this?

https://github.com/woocommerce/woocommerce/blob/6151f4021c7b0beb9b2effcb41fdb6b18350655b/plugins/woocommerce-admin/client/payments/payment-recommendations.tsx#L84-L94

**Review by chihsuan — COMMENTED** [review]
Tested well on my local for all scenarios. 👍  Just left some suggestions.

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (rjchow) [reply plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: Hmm... that's new to me... however I can't actually find the source for @woocommerce/settings and the only mention I've seen of it is in the changelog where it says its been removed

How does this work? 👀👀👀

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (rjchow) [reply plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: Yea good call! I was planning to extract the entire hook and everything but ended up realising I only needed a small part of the useSelect and forgot that there was this part below that's duplicated

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Tested well for me. Great work, @rjchow! The only thing left I think is to finalize the experiment name if we intend to do it dynamically.

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (chihsuan) [reply plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: Hmm...I think it's still injected into our code via the Dependency Extraction Webpack Plugin, though I also don't know how it exactly works.

Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: >however I can't actually find the source for @woocommerce/settings and the only mention I've seen of it is in the changelog where it says its been removed

The settings package used to be under WCAdmin, but now lives in blocks https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/trunk/assets/js/settings/shared

P2: p90Yrv-2B5-p2

**Comment by ilyasfoo** [comment]
I'll be working on my own feedback soon. @chihsuan, would you like to address your feedback as well or would you think it's better to address it as a follow-up?

**Comment by chihsuan** [comment]
> I'll be working on my own feedback soon. @chihsuan, would you like to address your feedback as well or would you think it's better to address it as a follow-up?

@ilyasfoo Thanks for the ping. Since those are just small changes, I'll address my feedback today!

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (chihsuan) [reply plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: Changed in 3ee2fcaeac2a62e92a40ebe0c17f37b90d0fe422

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (chihsuan) [reply plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: Updated in 11f321a334220dd296e999426010702221aaba73

**Comment by ilyasfoo** [comment]
@chihsuan Nice! I'm done with my changes. Perhaps @adrianduffell or @moon0326 can have a final look before merging after @chihsuan is done

File: plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx
Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/payments/payment-settings-banner.tsx]: Done in [2398fdf](https://github.com/woocommerce/woocommerce/pull/32697/commits/2398fdf772956a98d30317d000a9d7250a9b01be)

**Review by adrianduffell — DISMISSED** [review]
LGTM! 🚀 

Nice work on this @rjchow and thanks for helping this over the line @chihsuan @ilyasfoo ⭐

The "Payments Methods" header flash of text looks a bit janky to me, and I have concerns that removing the header makes it inconsistent with the other tabs. However we could follow-up on that in another PR after getting design feedback about it.

**Comment by adrianduffell** [comment]
@ilyasfoo I think we need to dismiss your review to merge? 🤔

**Review by ilyasfoo — DISMISSED** [review]

**Comment by ilyasfoo** [comment]
>The "Payments Methods" header flash of text looks a bit janky to me, and I have concerns that removing the header makes it inconsistent with the other tabs. However we could follow-up on that in another PR after getting design feedback about it.

Sounds good!

>@ilyasfoo I think we need to dismiss your review to merge? 🤔

Oops sorry missed this, I've approved it instead!

**Comment by ilyasfoo** [comment]
@adrianduffell do you think we should attempt to merge from trunk to see if it fixes checks? Or should we just merge it anyway?

**Comment by adrianduffell** [comment]
> do you think we should attempt to merge from trunk to see if it fixes checks?

@ilyasfoo Ah yep, I think we should run the checks. I'll push a commit with the merge from trunk.

**Comment by adrianduffell** [comment]
Looks like the e2e tests are failing on the payments settings screen. I'll take a closer look tomorrow.

**Comment by ilyasfoo** [comment]
@adrianduffell  I think that may be related to the "Payment Methods" header missing. I'll take a look instead, I suspect changing it to detect its table is better as RJ suggested

**Comment by ilyasfoo** [comment]
I think I've fixed e2e with d94769982b8a571197018ba4e5a4d34ca8f700af, but it seems some checks are failing due to unrelated errors. I'm not sure yet what it is, but it may have come with merge from `trunk`.

**Review by ilyasfoo — APPROVED** [review]

**Comment by ilyasfoo** [comment]
Checks are successful now, merging

**Comment by ilyasfoo** [comment]
Checks are successful now, merging

**Pull request merged**
